src/frontends/src/query_frontend.hpp

```diff
@@ -228,15 +170,19 @@ class RequestHandler {
       std::string json_content, std::string name,
       std::vector<VersionedModelId> models, std::string policy,
       long latency_slo_micros, InputType input_type) {
-    std::istringstream is(json_content);
-    ptree pt;
-    read_json(is, pt);
+    rapidjson::Document d;
+    rapidjson::ParseResult ok = d.Parse(json_content.c_str());
+    if (!ok) {
+      std::stringstream ss;
+      ss << "JSON parse error: " << rapidjson::GetParseError_En(ok.Code())
+         << " (offset " << ok.Offset() << ")\n";
+      throw json_parse_error(ss.str());
+    }
 
-    long uid = pt.get<long>("uid");
-    std::shared_ptr<Input> input = decode_input(input_type, pt);
+    long uid = getLong(d, "uid");
```

Comment:
First instance of parsing function reuse outside of `json_util.hpp`. The idea I mentioned previously is that you can simply call `getLong` or `getIntArray` without repeating the error-handling logic.